src/py/mat3ra/made/tools/build/utils.py

```diff
@@ -78,10 +81,74 @@ def merge_materials(
     distance_tolerance: float = 0.01,
     merge_dangerously=False,
 ) -> Material:
+    """
+    Merge multiple materials into a single material.
+
+    If some of the atoms are considered too close within a tolerance, only the last atom is kept.
+
+    Args:
+        materials (List[Material]): List of materials to merge.
+        material_name (Optional[str]): Name of the merged material.
+        distance_tolerance (float): The tolerance to replace atoms that are considered too close with respect
+            to the coordinates in the last material in the list, in angstroms.
+        merge_dangerously (bool): If True, the lattices are merged "as is" with no sanity checks.
+    Returns:
+        Material: The merged material.
+    """
     merged_material = materials[0]
     for material in materials[1:]:
         merged_material = merge_two_materials(
             merged_material, material, material_name, distance_tolerance, merge_dangerously
         )
 
     return merged_material
+
+
+def merge_two_materials_laterally(
+    phase_1_material: Material,
+    phase_2_material: Material,
+    gap: float,
+    cut_tolerance: Optional[float] = None,
+    distance_tolerance: float = 1.0,
+) -> Material:
+    """
+    Merge two materials laterally with translation along x axis with a gap between them.
+    Args:
+        phase_1_material (Material): The first material.
+        phase_2_material (Material): The second material.
+        gap (float): The gap between the two materials, in angstroms.
+        cut_tolerance (float): The cut tolerance to include atoms on the edge of cut, in crystal coordinates.
+        distance_tolerance (float): The distance tolerance to remove atoms that are too close, in angstroms.
+
+    Returns:
+        Material: The merged material.
+    """
+    # Default cut tolerance is half of the gap to allow atoms to fill the gap if they are on the edge
+    cut_tolerance = cut_tolerance or phase_1_material.basis.cell.convert_point_to_cartesian([gap, 0, 0])[2] / 2
```

Comment:
1. Should be taking the [0] component
2. Should be `convert_to_crystal`
3. cut_tolerance_in_crystal_units

Reply:
This logic seems to be off, I added the direct tolerance in angstroms: a hangover from the edge, it shouldn't depend on gap, only on bonds/radii